### PR TITLE
Node attestor extension

### PR DESCRIPTION
**Pull request by martincapello**
The intention of this PR is to begin extending server's node attestation by introducing a new optional list of selectors to nodeattestor.AttestResponse, which would allow to specify additional data (selectors) that can be used to resolve a node's identity.
Specifically this focuses on making x509pop node attestor to leverage this extensions. The idea is to be able to get the leaf certificate's Subject.CommonName and a fingerprint per each certificate of any of the  verified certificates chains and use that data on the server to create node resolver map entries.

### CODE REVIEW

File: pkg/server/endpoints/node/handler.go [hunk 1]
```diff
@@ -417,15 +416,17 @@ func (h *Handler) updateNodeResolverMap(ctx context.Context,
...
-				},
+			err := h.createNodeResolverMapEntry(ctx, baseSpiffeID, selector)
+			if err != nil {
+				return err
 			}
-			_, err = dataStore.CreateNodeResolverMapEntry(ctx, mapEntryRequest)
+		}
+	}
+
+	if attestResponse.Selectors != nil {
```
Comment (azdagron) [inline pkg/server/endpoints/node/handler.go]: this if check doesn't seem necessary

Comment (martincapello) [reply pkg/server/endpoints/node/handler.go]: Thank you!

File: pkg/server/plugin/nodeattestor/x509pop/x509pop.go [hunk 2]
```diff
@@ -121,6 +122,7 @@ func (p *X509PoPPlugin) Attest(stream nodeattestor.Attest_PluginStream) error {
 	resp := &nodeattestor.AttestResponse{
 		Valid:        true,
 		BaseSPIFFEID: x509pop.SpiffeID(c.trustDomain, leaf),
+		Selectors:    buildSelectors(leaf, chains),
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/x509pop/x509pop.go]: would you mind updating the unit-tests to assert the correct selectors are being returned?

File: pkg/server/plugin/nodeattestor/x509pop/x509pop.go [hunk 3]
```diff
@@ -175,3 +177,34 @@ func (p *X509PoPPlugin) setConfiguration(c *configuration) {
 func newError(format string, args ...interface{}) error {
 	return fmt.Errorf("x509pop: "+format, args...)
 }
+
+func buildSelectors(leaf *x509.Certificate, chains [][]*x509.Certificate) []*common.Selector {
+	selectors := []*common.Selector{}
+
+	if leaf.Subject.CommonName != "" {
+		selectors = append(selectors, &common.Selector{
+			Type: "x509", Value: "subject:cn:" + leaf.Subject.CommonName,
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/x509pop/x509pop.go]: Hmm, should Type be "x509" or "x509pop"?

File: pkg/server/plugin/nodeattestor/x509pop/x509pop.go (hunk 2 above)
Comment (martincapello) [reply pkg/server/plugin/nodeattestor/x509pop/x509pop.go]: Sure! Sorry about that

**Pull request merged**